src/peer_discovery.cpp

```diff
@@ -149,22 +162,60 @@ PeerDiscovery::DomainPeerDiscovery::loopListener()
     });
 }
 
+void
+PeerDiscovery::DomainPeerDiscovery::query(const asio::ip::udp::endpoint peer)
+{
+    std::lock_guard<std::mutex> lck(mtx_);
+    if (not lrunning_)
+        return;
+
+    msgpack::sbuffer pbuf_request;
+    std::map<std::string, std::string> request;
+    request["q"] = "id";
+    msgpack::pack(pbuf_request, request);
+
+    sockFd_.async_send_to(asio::buffer(pbuf_request.data(), pbuf_request.size()), peer,
+            [logger=logger_] (const asio::error_code& ec, size_t)
+            {
+                if (ec and (ec != asio::error::operation_aborted) and logger)
+                    logger->w("Error sending packet: %s", ec.message().c_str());
+            }
+    );
+}
+
 void
 PeerDiscovery::DomainPeerDiscovery::publish()
 {
     std::lock_guard<std::mutex> lck(mtx_);
-    if (lrunning_) {
-        sockFd_.async_send_to(asio::buffer(sbuf_.data(), sbuf_.size()), sockAddrSend_, [logger=logger_](const asio::error_code& ec, size_t){
-            if (ec and ec != asio::error::operation_aborted) {
-                if (logger)
+    if (not lrunning_)
+        return;
+
+    sockFd_.async_send_to(asio::buffer(sbuf_.data(), sbuf_.size()), sockAddrSend_,
+            [logger=logger_] (const asio::error_code& ec, size_t)
+            {
+                if (ec and (ec != asio::error::operation_aborted) and logger)
                     logger->w("Error sending packet: %s", ec.message().c_str());
             }
-        });
-        publishTimer_.expires_after(3s);
-        publishTimer_.async_wait(std::bind(&PeerDiscovery::DomainPeerDiscovery::publish, this));
-    }
+    );
+}
+
+void
+PeerDiscovery::DomainPeerDiscovery::publish(const asio::ip::udp::endpoint peer)
+{
+    std::lock_guard<std::mutex> lck(mtx_);
+    if (not lrunning_)
+        return;
+
+    sockFd_.async_send_to(asio::buffer(sbuf_.data(), sbuf_.size()), peer,
+            [logger=logger_] (const asio::error_code& ec, size_t)
+            {
+                if (ec and (ec != asio::error::operation_aborted) and logger)
+                    logger->w("Error sending packet: %s", ec.message().c_str());
+            }
+    );
```

Comment:
duplicated code, original publish could simply be given sockAddrSend_ as argument